server/auth/store.go

```diff
@@ -571,7 +578,11 @@ func (as *authStore) UserGet(r *pb.AuthUserGetRequest) (*pb.AuthUserGetResponse,
 }
 
 func (as *authStore) UserList(r *pb.AuthUserListRequest) (*pb.AuthUserListResponse, error) {
-	users := as.be.GetAllUsers()
+	tx := as.be.BatchTx()
```

Comment:
Please use` as.be.ReadTx() `instead if it's readonly operation.

Reply:
Thanks for pointing out, I mistakenly used `BatchTx`. But after changing it to `ReadTx`, I saw below error of the unit test:
```
--- FAIL: TestGetAllUsers (0.37s)
    --- FAIL: TestGetAllUsers/Empty_by_default (0.37s)
panic: do not use unsafeRange on non-keys bucket [recovered]
	panic: do not use unsafeRange on non-keys bucket
```

It's caused by this line: https://github.com/etcd-io/etcd/blob/main/server/storage/backend/read_tx.go#L85

This is the reason buckets other than the one for keys shouldn't allow unsafe range: https://github.com/etcd-io/etcd/blob/0bac6096325c20b0709a3565fbf5808aa7fbf917/server/storage/backend/batch_tx.go#L39

I'm still not sure read txns actually see anomalies of duplicated keys. But probably it's safe to use BatchTx for `AuthBackend.GetAllUsers()`. How do you think?

Reply:
I think we might need to use [UnsafeForEach](https://github.com/etcd-io/etcd/blob/0bac6096325c20b0709a3565fbf5808aa7fbf917/server/storage/backend/read_tx.go#L53) in this case.

Reply:
The range operation only makes sense for the key space/buckets, I think it should be the reason why the original author introduces the restriction. 

For non-key space/buckets, users are supposed to perform a single key/value lookup or ListAll operation. For ListAll, users should use [UnsafeForEach](https://github.com/etcd-io/etcd/blob/0bac6096325c20b0709a3565fbf5808aa7fbf917/server/storage/backend/read_tx.go#L53)

Reply:
Yeah I see, let me examine `UnsafeForEach` and update this PR.